indicators/talib_test.go

```diff
@@ -0,0 +1,771 @@
+	expectedHlc3 := []float64{7542.743333333333, 7495.649999999999, 7424.849999999999, 7255.433333333333, 7200.233333333333, 7185.603333333333, 7246.490000000001, 7111.836666666666, 7115.7699999999995, 6950.033333333333, 6700.356666666667, 7053.863333333334, 7173.8966666666665, 7161.61, 7150.420000000001, 7388.073333333334, 7419.003333333334, 7279.540000000001, 7191.110000000001, 7256.766666666666, 7184.456666666666, 7292.296666666666, 7394.81, 7268.046666666666, 7194.420000000001, 7188.676666666666, 7011.833333333333, 7198.25, 7332.203333333334, 7387.233333333334, 7640.733333333333, 8025.346666666667, 8126.7, 7864.563333333333, 8022.333333333333, 8100.740000000001, 8110.253333333334, 8113.606666666667, 8604.35, 8755.79, 8712.136666666667, 8856.456666666665, 8902.51, 8782.026666666667, 8624.546666666667, 8658.85, 8672.196666666667, 8441.463333333333, 8388.576666666666, 8339.896666666666, 8488.9, 8815.146666666667, 9229.746666666666, 9316.213333333333, 9412.023333333333, 9351.043333333333, 9373.886666666665, 9312.89, 9371.386666666667, 9194.323333333334, 9511.980000000001, 9711.733333333332, 9796.12, 9836.516666666666, 10079.183333333334, 9927.020000000002, 10119.94, 10363.71, 10265.723333333333, 10285.1, 10015.606666666667, 9854.206666666667, 9712.723333333333, 10025.256666666666, 9735.906666666668, 9571.980000000001, 9677.31, 9653.01, 9884.456666666667, 9720.08, 9407.446666666665, 8925.710000000001, 8770.416666666666, 8675.423333333334, 8619.266666666665, 8565.02, 8794.8, 8779.096666666666, 8755.68, 8997.813333333334, 9110.550000000001, 8990.85, 8310.1, 7921.126666666667, 7926.110000000001, 7838.446666666667, 5862.72, 5154.363333333334, 5294.46, 5476.903333333333}
+
+	for i := range expectedHlc3 {
+		if hlc3[i] != expectedHlc3[i] {
+			t.Fatalf("Assertion error: expected %f, got %f", expectedHlc3[i], hlc3[i])
+		}
+	}
+}
+
+// func TestCrossover(t *testing.T) {
+// 	if Crossover(testCrossunder1, testCrossunder2) == true {
+// 		t.Error("Crossover: Not expected and found")
+// 	}
+//
+// 	if Crossover(testNothingCrossed1, testNothingCrossed2) == true {
+// 		t.Error("Crossover: Not expected and found")
+// 	}
+//
+// 	if Crossover(testCrossover1, testCrossover2) == false {
+// 		t.Error("Crossover: Expected and not found")
+// 	}
+// }
+
+// func TestCrossunder(t *testing.T) {
+// 	// Crossunder
+// 	series1 := []float64{1, 2, 3, 4, 8, 6, 7}
+// 	series2 := []float64{1, 1, 10, 9, 5, 3, 7}
+
+// 	if Crossunder(series1, series2) == false {
+// 		t.Error("Crossunder: Expected and not found")
+// 	}
+
+// 	// Nothing
+// 	series1 = []float64{1, 2, 3, 4, 8, 6, 7}
+// 	series2 = []float64{1, 4, 5, 9, 5, 3, 7}
+
+// 	if Crossunder(series1, series2) == true {
+// 		t.Error("Crossunder: Not expected and found")
+// 	}
+
+// 	// Crossover
+// 	series1 = []float64{1, 3, 2, 4, 8, 6, 7}
+// 	series2 = []float64{1, 5, 1, 4, 5, 6, 7}
+
+// 	if Crossunder(series1, series2) == true {
+// 		t.Error("Crossunder: Not expected and found")
+// 	}
+// }
```

Comment:
reinstate?

Reply:
:+1: forgot to update with new data will do that now